core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnOrigins.java

```diff
@@ -303,6 +328,17 @@ private static Set<RelColumnOrigin> getMultipleColumns(RexNode rexNode, RelNode
             }
             return null;
           }
+
+          @Override public Void visitFieldAccess(RexFieldAccess fieldAccess) {
```

Comment:
shouldn't this actually use as origin the table that correlated variable comes from (recursively)?

Reply:
I'm not sure I fully understand what you mean. There is a case where the RelNode that calls getColumnOrigin is not the top-level RelNode, so it may not get the original table.